service/history/api/signalwithstartworkflow/api.go

```diff
@@ -71,19 +71,25 @@ func Invoke(
 		return nil, err
 	}
 
-	// Start workflow and signal
+	api.MigrateWorkflowIdDuplicationPolicies(
+		&signalWithStartRequest.SignalWithStartRequest.WorkflowIdReusePolicy,
+		&signalWithStartRequest.SignalWithStartRequest.WorkflowIdConflictPolicy)
+
```

Comment:
I decided to do the migration here instead of in Frontend because that makes it more reliable when running a mix of old/new versions across both services (as it could be in the case of a deployment).